aci/resource_aci_bfdmhifp.go

```diff
@@ -0,0 +1,305 @@
+
+	bfdMhIfPAttr := models.AciBfdMultihopInterfaceProfileAttributes{}
+
+	nameAlias := ""
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		nameAlias = NameAlias.(string)
+	}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		bfdMhIfPAttr.Annotation = Annotation.(string)
+	} else {
+		bfdMhIfPAttr.Annotation = "{}"
+	}
+
+	if Key, ok := d.GetOk("key"); ok {
+		bfdMhIfPAttr.Key = Key.(string)
+	}
+
+	if KeyId, ok := d.GetOk("key_id"); ok {
+		bfdMhIfPAttr.KeyId = KeyId.(string)
+	}
+
+	if Name, ok := d.GetOk("name"); ok {
+		bfdMhIfPAttr.Name = Name.(string)
+	}
+
+	if InterfaceProfile_type, ok := d.GetOk("interface_profile_type"); ok {
+		bfdMhIfPAttr.InterfaceProfile_type = InterfaceProfile_type.(string)
+	}
+	bfdMhIfP := models.NewAciBfdMultihopInterfaceProfile(fmt.Sprintf("bfdMhIfP"), LogicalInterfaceProfileDn, desc, nameAlias, bfdMhIfPAttr)
+
+	bfdMhIfP.Status = "modified"
+
+	err := aciClient.Save(bfdMhIfP)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	checkDns := make([]string, 0, 1)
+
+	if d.HasChange("relation_bfd_rs_mh_if_pol") || d.HasChange("annotation") {
+		_, newRelParam := d.GetChange("relation_bfd_rs_mh_if_pol")
+		checkDns = append(checkDns, newRelParam.(string))
+
+	}
+
+	d.Partial(true)
+	err = checkTDn(aciClient, checkDns)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.Partial(false)
+
+	if d.HasChange("relation_bfd_rs_mh_if_pol") || d.HasChange("annotation") {
+		_, newRelParam := d.GetChange("relation_bfd_rs_mh_if_pol")
+		err = aciClient.DeleteRelationbfdRsMhIfPol(bfdMhIfP.DistinguishedName)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		err = aciClient.CreateRelationbfdRsMhIfPol(bfdMhIfP.DistinguishedName, bfdMhIfPAttr.Annotation, GetMOName(newRelParam.(string)))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+	}
+
+	d.SetId(bfdMhIfP.DistinguishedName)
+	log.Printf("[DEBUG] %s: Update finished successfully", d.Id())
+	return resourceAciBfdMultihopInterfaceProfileRead(ctx, d, m)
+}
+
+func resourceAciBfdMultihopInterfaceProfileRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] Aci BFD Multihop Interface Profile %s: Beginning Read", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+
+	bfdMhIfP, err := getRemoteAciBfdMultihopInterfaceProfile(aciClient, dn)
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+
+	_, err = setAciBfdMultihopInterfaceProfileAttributes(bfdMhIfP, d)
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+
+	bfdRsMhIfPolData, err := aciClient.ReadRelationbfdRsMhIfPol(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation bfdRsMhIfPol %v", err)
+		d.Set("relation_bfd_rs_mh_if_pol", "")
+	} else {
+		if _, ok := d.GetOk("relation_bfd_rs_mh_if_pol"); ok {
+			tfName := GetMOName(d.Get("relation_bfd_rs_mh_if_pol").(string))
+			if tfName != bfdRsMhIfPolData {
+				d.Set("relation_bfd_rs_mh_if_pol", "")
+			}
+		}
+	}
```

Comment:
dont think we need the else

```suggestion
}

if _, ok := d.GetOk("relation_bfd_rs_mh_if_pol"); ok {
	tfName := GetMOName(d.Get("relation_bfd_rs_mh_if_pol").(string))
	if tfName != bfdRsMhIfPolData {
		d.Set("relation_bfd_rs_mh_if_pol", "")
	}
}

```

Reply:
nope removed